ksqldb-test-util/src/main/java/io/confluent/ksql/test/util/TestMethods.java

```diff
@@ -65,13 +67,57 @@ private TestMethods() {
   public static final class Builder<T> {
 
     private final Class<T> typeUnderTest;
-    private final Set<Method> blackList = new HashSet<>();
+    private final Set<MethodRef> blackList = new HashSet<>();
     private final Map<Class<?>, Object> defaults = new HashMap<>();
-
     public Builder(final Class<T> typeUnderTest) {
       this.typeUnderTest = Objects.requireNonNull(typeUnderTest, "typeUnderTest");
     }
 
+    /**
+     * The purpose of this class is to provide a more general reference to the actual Method that we can
+     * use in the blacklist. Note that the equality algorithm is the same as Method's: class name, method name,
+     * and argument types. I expect this to be a valid equality algorithm for methods indefinitely because it's
+     * the same as the method overload resolution criteria specified in the Java language.
+     */
+    public static class MethodRef {
```

Comment:
As the doc says, I'm just creating this little "bean" class so I can blacklist methods that don't actually exist on the declaring class (as well as ones that do). Regardless, the "contains"/"equality" criteria are the same as those of the real Method.